src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -151,7 +151,23 @@ public QueryLogsResults queryLogs(QueryLogsParams queryLogsParams) throws LogSer
                     new GeneralLogEntry.SourceLocation(sourceLocation.getFile(), sourceLocation.getLine(),
                             sourceLocation.getFunction()), timestamp);
             if (payload.getType() == Payload.Type.JSON) {
-                Map<String, Object> jsonPayloadMap = ((Payload.JsonPayload) payload).getDataAsMap();
+                Map<String, Object> jsonPayloadMap = new HashMap<>(((Payload.JsonPayload) payload).getDataAsMap());
+                if (!isUserAdmin) {
```

Comment:
This is not the right place to do it. Hidden information needs to be hidden regardless of whether the log comes from Cloud Logging or later on local development.

Reply:
Understood. It is updated.